lib/HTTP/Cookies.pm

```diff
@@ -621,6 +621,50 @@ sub _normalize_path  # so that plain string compare can be used
     $_[0] =~ s/([\0-\x20\x7f-\xff])/sprintf("%%%02X",ord($1))/eg;
 }
 
+# deals with splitting values by ; and the fact that they could
+# be in quotes which can also have escaping.
+sub _split_text {
+    my $val = shift;
+    my @vals = grep { $_ ne '' } split(/([;\\"])/, $val);
```

Comment:
While I agree that the alternative quoting can be useful for readability at times I'm not convinced in this case.  I'm not having to escape anything excessively because of a clash of quotes, and I don't see any particular precedence for this being used in the existing code base that makes me think I should do it to fit in.

If anything I think it would make it harder to grok for those less familiar with the slightly more arcane (but useful) parts of perl.

I will make the change if you insist, but I'm not completely convinced of the merits.